src/Symfony/Component/PasswordHasher/Tests/Hasher/NativePasswordHasherTest.php

```diff
@@ -98,17 +98,37 @@ public function testBcryptWithLongPassword()
         $this->assertTrue($hasher->verify($hasher->hash($plainPassword), $plainPassword));
     }
 
+    /**
+     * @requires PHP < 8.4
+     */
     public function testBcryptWithNulByte()
     {
         $hasher = new NativePasswordHasher(null, null, 4, \PASSWORD_BCRYPT);
         $plainPassword = "a\0b";
 
-        if (\PHP_VERSION_ID < 80218 || \PHP_VERSION_ID >= 80300 && \PHP_VERSION_ID < 80305) {
-            // password_hash() does not accept passwords containing NUL bytes since PHP 8.2.18 and 8.3.5
-            $this->assertFalse($hasher->verify(password_hash($plainPassword, \PASSWORD_BCRYPT, ['cost' => 4]), $plainPassword));
+        try {
+            $hash = password_hash($plainPassword, \PASSWORD_BCRYPT, ['cost' => 4]);
+        } catch (\Throwable $throwable) {
+            // we skip the test in case the PHP version does not support NUL bytes in passwords
+            // with bcrypt
+            //
+            // @see https://github.com/php/php-src/commit/11f2568767660ffe92fbc6799800e01203aad73a
+            if (false !== strpos($throwable->getMessage(), 'Bcrypt password must not contain null character')) {
+                $this->markTestSkipped('password_hash() does not accept passwords containing NUL bytes.');
```

Comment:
We should not skip the test in this case (and the case below) as we still want to apply the `hash()` method shipped with the Symfony component which is able to handle this case gracefully.

But, since this seems to cause some confusion we should maybe split this test into two.

Reply:
Ah yes I didn't catch it was kind of 2 tests in one. I'll update accordingly.